vector/src/main/java/im/vector/app/features/voicebroadcast/listening/VoiceBroadcastPlayerImpl.kt

```diff
@@ -416,6 +432,10 @@ class VoiceBroadcastPlayerImpl @Inject constructor(
         }
 
         override fun onCompletion(mp: MediaPlayer) {
+            // Release media player as soon as it completed
+            mp.release()
+            currentMediaPlayer = null
```

Comment:
Should we check `mp` and `currentMediaPlayer` are referencing the same instance before setting it to `null`?

Reply:
Not sure, what should I do if it is not the same instance? In that case, we'll be in an unexpected state. Should we stop the listening, throw an exception, release the other instance...?

Reply:
If it is not expected, I would say we can throw error like this:
```kotlin
mp.release()
if (currentMediaPlayer == mp) {
    currentMediaPlayer = null
} else {
    throw IllegalStateException("Message to explain the situation.")
}
```
What do you think?
In the general case it will never throw except if the implementation is not correct, am I right?

Reply:
Done: https://github.com/vector-im/element-android/pull/7919/commits/169c9b221ca69039406a50744749c5336c4af85a
I did some tests and the completion is triggered before the next player starts so it should be fine.